pandas/core/series.py

```diff
@@ -3814,9 +3819,13 @@ def explode(self) -> "Series":
 
         values, counts = reshape.explode(np.asarray(self.array))
 
-        result = self._constructor(
-            values, index=self.index.repeat(counts), name=self.name
-        )
+        if ignore_index:
+            result = self._constructor(values, index=range(len(values)), name=self.name)
```

Comment:
I don't think you need to add name here? do we do it for sort_values for a Series?

Reply:
`ibase.default_index` is used, same as I used in `DataFrame.explode`, but that is not preferable here right, since we use the constructor, or should I use:

```
result = self._constructor(values, index=ibase.default_index(len(values)))
```

Reply:
@jreback it's the name of the Series, not Index. So that's certianly something we want to preserve I think (as we do right now in explode)

Reply:
we don’t do this for Dataframe and we already name the series correctly

this might be a bug from the origina impl

Reply:
> we already name the series correctly

indeed, repeating myself: "it's the name of the Series, not Index"

The ` name=self.name` in the line above is the name of the Series, which we want to preserve

Reply:
The other points are resolved, if this point is agreed upon I will push the changes.